packages/openapi3/src/cli/actions/convert/generators/common.ts

```diff
@@ -7,3 +7,18 @@ export function stringLiteral(value: string): string {
   // Escape both quotes and ${...} in single-line strings to prevent interpolation
   return `"${value.replaceAll('"', '\\"').replaceAll("${", "\\${")}"`;
 }
+
+/**
+ * Render a TypeSpec string literal for use in object literal values.
+ * Always uses escaped single-line format to avoid issues with nested triple-quotes.
+ */
+export function escapedStringLiteral(value: string): string {
```

Comment:
what is the difference with the function above?

Reply:
I just pushed changes to the main method, let me know if you need me to change anything else.